src/wxvx/tests/test_workflow.py

```diff
@@ -418,20 +418,41 @@ def mock(*_args, **_kwargs):
     }
 
 
-@mark.parametrize(
-    "template",
-    [
-        "{root}/gfs.t00z.pgrb2.0p25.f000",
-        "file://{root}/gfs.t00z.pgrb2.0p25.f000",
-    ],
-)
-def test_workflow__grid_grib__local(template, config_data, gen_config, fakefs, tc, testvars):
-    grib_path = fakefs / "gfs.t00z.pgrb2.0p25.f000"
-    grib_path.write_text("foo")
+def test_workflow__grib_index_ec(c, fakefs, tc):
+    grib = fakefs / "foo"
+    grib.touch()
+    with patch.object(workflow, "ec") as ec:
+        ec.codes_index_write.side_effect = lambda _idx, p: Path(p).touch()
+        assert ready(workflow._grib_index_ec(c=c, grib_path=grib, tc=tc))
+    yyyymmdd, hh, leadtime = tcinfo(tc)
+    idx_path = c.paths.grids_baseline / yyyymmdd / hh / leadtime / f"{grib.name}.ecidx"
+    assert idx_path.is_file()
+
+
+@mark.parametrize(("msgs", "expected"), [(0, False), (1, True), (2, True)])
+def test__workflow__grib_message_in_file(c, expected, fakefs, msgs, tc, testvars):
+    grib_path = fakefs / "foo"
+    idx_path = fakefs / "foo.ecidx"
+    idx_path.touch()
+    with (
+        patch.object(workflow, "_grib_index_ec") as _grib_index_ec,
+        patch.object(workflow, "ec") as ec,
+    ):
+        ec.codes_new_from_index.side_effect = [object()] * msgs + [None]
+        node = workflow._grib_message_in_file(c=c, path=grib_path, tc=tc, var=testvars["gh"])
+        assert node.ready is expected
```

Comment:
```suggestion
        assert node.ready is expected
        if msgs == 2:
            assert logged("Found 2 GRIB messages")
```
This might not be exactly right, but hopefully close. I think it would be nice to test that the warning message is logged if we get more than 1 match. The `logged` fixture will also need to be added to the function's arguments.